src/main/java/com/ft/methodearticlemapper/resources/PostContentToTransformResource.java

```diff
@@ -39,42 +48,58 @@ public PostContentToTransformResource(EomFileProcessor eomFileProcessor) {
 		final String uuid = eomFile.getUuid();
 		validateUuid(uuid);
-		return processRequest(uuid, preview, eomFile, transactionId);
+		
+		TransformationMode transformationMode = null;
+		if (mode == null) {
+		    transformationMode = preview ? TransformationMode.PREVIEW : TransformationMode.PUBLISH;
+		} else {
+		    try {
+		        transformationMode = TransformationMode.valueOf(mode.toUpperCase());
+		    } catch (IllegalArgumentException e) {
+	            throw ClientError.status(SC_BAD_REQUEST)
+                                 .context(uuid)
+                                 .error(e.getMessage())
+                                 .exception(e);
+		    }
+		}
+        // otherwise, mode trumps the preview flag if there is a mismatch
+		
+		return processRequest(uuid, transformationMode, eomFile, transactionId);
 	}
 
-    private Content processRequest(String uuid, boolean preview, EomFile eomFile, String transactionId) {
+    private Content processRequest(String uuid, TransformationMode mode, EomFile eomFile, String transactionId) {
         try {
-            Date lastModifiedDate = new Date();
-            if (preview) {
-                return eomFileProcessor.processPreview(eomFile, transactionId, lastModifiedDate);
-            }
-            return eomFileProcessor.processPublication(eomFile, transactionId, lastModifiedDate);
+            return eomFileProcessor.process(eomFile, mode, transactionId, new Date());
 
         } catch (MethodeMarkedDeletedException e) {
-            throw ClientError.status(404)
+            throw ClientError.status(SC_NOT_FOUND)
                     .context(uuid)
                     .reason(ErrorMessage.METHODE_FILE_NOT_FOUND)
                     .exception(e);
         } catch (NotWebChannelException e) {
-            throw ClientError.status(422)
+            throw ClientError.status(SC_UNPROCESSABLE_ENTITY)
                     .reason(ErrorMessage.NOT_WEB_CHANNEL)
                     .exception(e);
         } catch (MethodeMissingFieldException e) {
-            throw ClientError.status(422)
+            throw ClientError.status(SC_UNPROCESSABLE_ENTITY)
                     .error(String.format(ErrorMessage.METHODE_FIELD_MISSING.toString(), e.getFieldName()))
                     .exception(e);
         } catch (MethodeMissingBodyException | UntransformableMethodeContentException e) {
-            throw ClientError.status(422)
+            throw ClientError.status(SC_UNPROCESSABLE_ENTITY)
                     .error(e.getMessage())
                     .exception(e);
         } catch (MethodeContentNotEligibleForPublishException e) {
-            throw ClientError.status(422)
+            throw ClientError.status(SC_UNPROCESSABLE_ENTITY)
+                    .context(uuid)
+                    .error(e.getMessage())
+                    .exception(e);
+        } catch (UnsupportedTransformationModeException e) {
+            throw ClientError.status(SC_MISDIRECTED_REQUEST)
```

Comment:
Yeah, I'm not sure about it either. I was trying to avoid the ubiquitous `400` or `404`. This happens if you request a mapping with a mode that isn't supported in the current deployment, so, in that sense, the client has misdirected their request. I think I'd go for `404` as a Plan B - there's nothing syntactically incorrect about the request, so maybe "not found" is a reasonable response for this case.